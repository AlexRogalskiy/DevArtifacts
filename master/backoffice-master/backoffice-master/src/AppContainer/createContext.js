import {
  create,
  SheetsRegistry,
} from 'jss'
import { jssPreset } from '@material-ui/core/styles'
import createGenerateClassName from '@material-ui/core/styles/createGenerateClassName'

// Configure JSS
const jss = create(jssPreset())
jss.options.createGenerateClassName = createGenerateClassName

export const sheetsManager = new Map()

export default function createContext() {
  return {
    jss,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager,
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
  }
}
