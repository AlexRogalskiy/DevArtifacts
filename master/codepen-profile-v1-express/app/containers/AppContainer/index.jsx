/**
 * External dependencies
 */
import React, { PropTypes, Component, cloneElement } from 'react';
import $ from 'jquery';

/**
 * Internal dependencies
 */
import ArrowUpIcon from '../../assets/img/icon-arrow-up.svg';
import { Container, Row, Column } from '../../components/Grid';
import Button from '../../components/Button';
import Header from '../../components/Header';
import Footer from '../../components/Footer';
import SuggestionsCard from '../../components/SuggestionsCard';
import TrendingCard from '../../components/TrendingCard';
import UserCard from '../../components/UserCard';

import { TabBar } from '../../components/Mobile';

/**
 * Local variables
 */
const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {};

class AppContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      backToTop: false,
      user: 'andytran',
      fixedHeader: false,
    };

    this._handleToTop = this._handleToTop.bind(this);
  }

  componentWillMount() {
    this.setState({
      windowWidth: window.innerWidth,
    });

    window.addEventListener('resize', this._handleResize.bind(this));
    window.addEventListener('scroll', this._onScroll.bind(this));
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this._handleResize);
    window.removeEventListener('scroll', this._onScroll);
  }

  _handleResize() {
    this.setState({
      windowWidth: window.innerWidth,
    });
  }

  _onScroll() {
    if (window.pageYOffset > 0 && !this.state.fixedHeader) {
      this.setState({
        fixedHeader: true,
      });
    } else if (window.pageYOffset <= 0 && this.state.fixedHeader) {
      this.setState({
        fixedHeader: false,
      });
    }

    if (window.pageYOffset > 300 && !this.state.backToTop) {
      this.setState({
        backToTop: true,
      });
    } else if (window.pageYOffset < 300 && this.state.backToTop) {
      this.setState({
        backToTop: false,
      });
    }
  }

  _handleToTop() {
    $('html, body').animate({ scrollTop: 0 });
  }

  render() {
    if (this.state.windowWidth <= 768) {
      return (
        <div className="app app--mobile">
          <Header
            className="header--mobile"
            fixed={this.state.fixedHeader}
            windowWidth={this.state.windowWidth}
          />
          {this.props.children && cloneElement(this.props.children, {
            user: this.state.user,
            windowWidth: this.state.windowWidth,
          })}
          <TabBar />
        </div>
      );
    }

    return (
      <div className="app">
        <Header fixed={this.state.fixedHeader} />
        <Container>
          <Row>
            <Column size={this.state.windowWidth >= 960 ? 3 : 4}>
              <UserCard user={this.state.user} />
              <TrendingCard />

              {this.state.windowWidth >= 960 ? '' : <SuggestionsCard />}
              {this.state.windowWidth >= 960 ? '' : <Footer />}
            </Column>
            <Column size={this.state.windowWidth >= 960 ? 6 : 8}>
              {this.props.children && cloneElement(this.props.children, {
                user: this.state.user,
                windowWidth: this.state.windowWidth,
              })}
            </Column>
            {this.state.windowWidth >= 960 ?
              <Column size={3}><SuggestionsCard /><Footer /></Column> : ''}
          </Row>
        </Container>
        <Button
          className={
            this.state.backToTop ?
              'button--back-to-top button--visible' :
              'button--back-to-top button--hidden'
          }
          onClick={this._handleToTop}
        >
          <span dangerouslySetInnerHTML={{ __html: ArrowUpIcon }} />
        </Button>
      </div>
    );
  }
}

AppContainer.propTypes = propTypes;
AppContainer.defaultProps = defaultProps;

export default AppContainer;
