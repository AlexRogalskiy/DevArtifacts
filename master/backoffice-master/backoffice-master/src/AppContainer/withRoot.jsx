import React, { Component } from 'react'
import JssProvider from 'react-jss/lib/JssProvider'
import wrapDisplayName from 'recompose/wrapDisplayName'

import { withStyles, MuiThemeProvider } from '@material-ui/core/styles'

import createContext from './createContext'

// Apply some reset
const styles = theme => ({
  '@global': {
    html: {
      background: theme.palette.background.default,
      WebkitFontSmoothing: 'antialiased', // Antialiasing.
      MozOsxFontSmoothing: 'grayscale', // Antialiasing.
    },
    body: {
      margin: 0,
    },
  },
})

let AppWrapper = props => props.children

AppWrapper = withStyles(styles)(AppWrapper)

const context = createContext()

const withRoot = (BaseComponent, props) => {
  class WithRoot extends Component {
    componentDidMount() {
      // Remove the server-side injected CSS.
      const jssStyles = document.querySelector('#jss-server-side')

      if (jssStyles && jssStyles.parentNode) {
        jssStyles.parentNode.removeChild(jssStyles)
      }
    }

    render() {
      return (
        <JssProvider registry={context.sheetsRegistry} jss={context.jss}>
          <MuiThemeProvider theme={props.theme} sheetsManager={context.sheetsManager}>
            <AppWrapper>
              <BaseComponent {...this.props} />
            </AppWrapper>
          </MuiThemeProvider>
        </JssProvider>
      )
    }
  }

  if (process.env.NODE_ENV !== 'production') {
    WithRoot.displayName = wrapDisplayName(BaseComponent, 'withRoot')
  }

  return WithRoot
}

export default withRoot
