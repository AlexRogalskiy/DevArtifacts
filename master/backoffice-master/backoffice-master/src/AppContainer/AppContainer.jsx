import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { createMuiTheme } from '@material-ui/core'
import {
  blueGrey,
  orange,
} from '@material-ui/core/colors'

import withRoot from './withRoot'

const defaultTheme = {
  palette: {
    primary: {
      light: blueGrey[300],
      main: blueGrey[500],
      dark: blueGrey[700],
    },
    secondary: {
      light: orange[300],
      main: orange[500],
      dark: orange[700],
    },
  },
}

const AppContainer = ({ theme, children }) => {
  const BaseComponent = () => (
    <Fragment>
      {children}
    </Fragment>
  )

  const Root = withRoot(BaseComponent, {
    theme: createMuiTheme(theme),
  })

  return (
    <Root />
  )
}

AppContainer.propTypes = {
  theme: PropTypes.objectOf(PropTypes.object),
  children: PropTypes.node.isRequired,
}

AppContainer.defaultProps = {
  theme: defaultTheme,
}

export default AppContainer
