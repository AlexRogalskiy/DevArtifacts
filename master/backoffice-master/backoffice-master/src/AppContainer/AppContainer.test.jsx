import React from 'react'
import { mount } from 'enzyme'

import AppContainer from './'

describe('Component Info', () => {
  it('renders correctly', () => {
    const tree = mount((
      <AppContainer>
        Foo
      </AppContainer>
    ))

    expect(tree).toMatchSnapshot()
  })
})
