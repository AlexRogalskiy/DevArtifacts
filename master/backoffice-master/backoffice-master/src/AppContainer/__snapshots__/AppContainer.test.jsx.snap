// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Info renders correctly 1`] = `"Foo"`;
